cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -86,8 +86,9 @@ class SnappyUnifiedMemoryManager private[memory](
 
   private val evictionFraction = SnappyUnifiedMemoryManager.getStorageEvictionFraction(conf)
 
-  private val maxHeapStorageSize =
-    (maxHeapMemory * evictionFraction).toLong
+  private val maxHeapStorageSize = (maxHeapMemory * evictionFraction).toLong
+
+  private val maxHeapExecutionSize = (maxHeapMemory * evictionFraction).toLong
```

Comment:
Do we want execution to grow into storage? I think it will be better to avoid that so maxHeapExecutionSize can be governed by "spark.memory.storageFraction" i.e. 50% by default. Also need to assert that storageFraction should be less than or equal to evictionFraction.